### PR TITLE
fix `git mv` commands when migrating files with `$` characters

### LINKED ISSUE #6113: Migrate tool passing unescaped shell commands?
### Describe the bug

Thank you for providing the migrate tool, I'm excited about the new routing strategy.

Migrating a project that reproduces some internet archive URLs that include the character "$" in them, I noticed that if a route path in the old format includes special characters, something odd happens that could looks like it could conceivably involve unescaped shell commands. I don't know how the large the scope is, but this seems like the sort of thing that could cause issues if someone has a route at `src/routes/page$[rm -rf /]` or possibly something remotely plausible.

I had a route at `src/routes/pages/[id]$[pagenum]/index.svelte`. (The format with `$` reproduces Internet Archive URLs).
Migrating throws an error `Command failed: git mv "src/routes/pages/[id]$[pagenum]/index.svelte"`  because `fatal: bad source, source=src/routes/pages/[id]0/index.svelte, destination=src/routes/pages/[id]0/+page.svelte`.  That is, the dollar sign triggers an [old bash syntax](https://unix.stackexchange.com/questions/209833/what-does-a-dollar-sign-followed-by-a-square-bracket-mean-in-bash) I've never heard of where `$[foo]` evaluates to '0'. Not the end of the world for me, but this seems like the sort of thing that in rare edge cases might be able cause some havoc on someone's machine, so I wanted to bring it to your attention.

### Reproduction

Create a path at `src/routes/pages/[id]$[pagenum]/index.svelte` in an old sveltekit project, and run npx svelte-migrate routes 

Sorry for not providing a full reprex, which is tricky because it requires forking an old thing.

### Logs

```shell
(base) benschmidt@Bens-MBP-2 ads % npx svelte-migrate routes

This will overwrite files in the current directory!

Your git working directory is dirty — we recommend committing your changes before running this migration.

✔ Continue? … yes
fatal: bad source, source=src/routes/pages/[id]0/index.svelte, destination=src/routes/pages/[id]0/+page.svelte
node:internal/errors:856
  const err = new Error(message);
              ^

Error: Command failed: git mv "src/routes/pages/[id]$[pagenum]/index.svelte" "src/routes/pages/[id]$[pagenum]/+page.svelte"
fatal: bad source, source=src/routes/pages/[id]0/index.svelte, destination=src/routes/pages/[id]0/+page.svelte

    at checkExecSyncError (node:child_process:820:11)
    at execSync (node:child_process:891:15)
    at move_file (file:///Users/benschmidt/.npm/_npx/ca732f115159e57c/node_modules/svelte-migrate/migrations/routes/utils.js:49:3)
    at migrate (file:///Users/benschmidt/.npm/_npx/ca732f115159e57c/node_modules/svelte-migrate/migrations/routes/index.js:163:4)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  status: 128,
  signal: null,
  output: [
    null,
    Buffer(0) [Uint8Array] [],
    Buffer(111) [Uint8Array] [
      102,  97, 116,  97, 108,  58,  32,  98,  97, 100,  32, 115,
      111, 117, 114,  99, 101,  44,  32, 115, 111, 117, 114,  99,
      101,  61, 115, 114,  99,  47, 114, 111, 117, 116, 101, 115,
       47, 112,  97, 103, 101, 115,  47,  91, 105, 100,  93,  48,
       47, 105, 110, 100, 101, 120,  46, 115, 118, 101, 108, 116,
      101,  44,  32, 100, 101, 115, 116, 105, 110,  97, 116, 105,
      111, 110,  61, 115, 114,  99,  47, 114, 111, 117, 116, 101,
      115,  47, 112,  97, 103, 101, 115,  47,  91, 105, 100,  93,
       48,  47,  43, 112,
      ... 11 more items
    ]
  ],
  pid: 94878,
  stdout: Buffer(0) [Uint8Array] [],
  stderr: Buffer(111) [Uint8Array] [
    102,  97, 116,  97, 108,  58,  32,  98,  97, 100,  32, 115,
    111, 117, 114,  99, 101,  44,  32, 115, 111, 117, 114,  99,
    101,  61, 115, 114,  99,  47, 114, 111, 117, 116, 101, 115,
     47, 112,  97, 103, 101, 115,  47,  91, 105, 100,  93,  48,
     47, 105, 110, 100, 101, 120,  46, 115, 118, 101, 108, 116,
    101,  44,  32, 100, 101, 115, 116, 105, 110,  97, 116, 105,
    111, 110,  61, 115, 114,  99,  47, 114, 111, 117, 116, 101,
    115,  47, 112,  97, 103, 101, 115,  47,  91, 105, 100,  93,
     48,  47,  43, 112,
    ... 11 more items
  ]
}

Node.js v18.4.0
```
```


### System Info

```shell
System:
    OS: macOS 11.6.1
    CPU: (8) arm64 Apple M1
    Memory: 183.20 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 18.4.0 - /opt/homebrew/bin/node
    npm: 8.12.1 - /opt/homebrew/bin/npm
  Browsers:
    Chrome: 104.0.5112.101
    Edge: 104.0.1293.63
    Firefox: 100.0.2
    Safari: 15.1
  npmPackages:
    @sveltejs/adapter-auto: ^1.0.0-next.64 => 1.0.0-next.64 
    @sveltejs/adapter-vercel: ^1.0.0-next.66 => 1.0.0-next.66 
    @sveltejs/kit: ^1.0.0-next.405 => 1.0.0-next.405 
    svelte: ^3.49.0 => 3.49.0
```

### Severity

serious, but I can work around it

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by Rich-Harris**
This should fix #6113 — tested with the case in that issue, works fine, though I'd appreciate a sanity check from others

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. All changesets should be `patch` until SvelteKit 1.0

### CODE REVIEW

**Review by dominikg — APPROVED** [review]

File: packages/migrate/migrations/routes/utils.js [hunk 1]
```diff
@@ -46,7 +47,7 @@ export function error(description, comment_id) {
  */
 export function move_file(file, renamed, content, use_git) {
 	if (use_git) {
-		execSync(`git mv ${JSON.stringify(file)} ${JSON.stringify(renamed)}`);
+		execSync(`git mv ${sq.quote([file, renamed])}`);
```
Comment (Conduitry) [inline packages/migrate/migrations/routes/utils.js:50]: Could we use https://nodejs.org/dist/latest/docs/api/child_process.html#child_processexecfilefile-args-options-callback instead so we can pass in an array of arguments directly and not have to worry about the shell escaping?

Comment (Conduitry) [reply packages/migrate/migrations/routes/utils.js:50]: Untested, but I think it'd be `execFileSync('git', ['mv', file, renamed])`.

Comment (dominikg) [reply packages/migrate/migrations/routes/utils.js:50]: ohh, thats nice.
Seems to work alright for `\n and $`
execFileSync:
```
node -e 'console.log(require("child_process").execFileSync("echo",["bla\\n\$1"]).toString())'
bla\n$1
```

execSync:
```
node -e 'console.log(require("child_process").execSync("echo bla\\n\$1").toString())'
blan
```

Comment (Rich-Harris) [reply packages/migrate/migrations/routes/utils.js]: ooh yeah that's way better

**Pull request merged**